### DISCUSSION TITLE
Unable to load schema from 'cmake-tools-schema:/schemas/CMakePresets-v4-schema.json': cannot open cmake-tools-schema:/schemas/CMakePresets-v4-schema.json. Detail: Unable to resolve resource cmake-tools-schema:/schemas/CMakePresets-v4-schema.json.

### DISCUSSION THREAD

**Discussion by urbans0ft** [microsoft/vscode-cmake-tools#2908] [Q&A]
I simply can't get `CMakePresets.json` to work.

- OS Version: 10.0.19041 N/A Build 19041
- vscode: 1.74.2
- CMake Tools: v1.12.27
- gcc.exe (Rev6, Built by MSYS2 project) 12.2.0
- cmake version 3.25.1

Besieds that I tested several other configurations e.g. with "Buildtools for Visual Studio", fresh vscode installations and different Laptops among other things. I'm lost. Vscode keeps telling me

```
Unable to load schema from 'cmake-tools-schema:/schemas/CMakePresets-v4-schema.json': cannot open cmake-tools-schema:/schemas/CMakePresets-v4-schema.json. Detail: Unable to resolve resource cmake-tools-schema:/schemas/CMakePresets-v4-schema.json.(768)
```

Every time I open up `CMakePresets.json`.

**Comment by bobbrow** [comment]
Sorry that you're having trouble with the schema.  We've seen reports of this before (#2587) but haven't been able to get it under a debugger.  VS Code owns the schema loading behavior and we've seen this issue come and go.  Sometimes reinstalling the extension will fix it.  It breaks IntelliSense for `CMakePresets.json`, but shouldn't stop you from using the presets otherwise.

**Reply by h-bex** [reply]
Perhaps this helps: If the CMakePresets.json is already open in the project when I start vscode the error occurs. In this case it does also not help to close the file and open it again.
If the CMakePresets.json file is not open in the project if I start vscode, I have no problem with the schema.

As a workaraound: Close the CMakePresets.json file in vscode; terminate vscode; start it again and open CMakePresets.json again. (NOTE: This works for my installation).
Environment:
Version: 1.77.1 (user setup)
Commit: b7886d7461186a5eac768481578c1d7ca80e2d21
Date: 2023-04-04T23:21:11.906Z
Electron: 19.1.11
Chromium: 102.0.5005.196
Node.js: 16.14.2
V8: 10.2.154.26-electron.0
OS: Windows_NT x64 10.0.22621
Sandboxed: Yes

CMake for VisualStudio Code:
Published
10/20/2015, 01:32:24
Last released
3/7/2017, 00:12:02
Last updated
10/25/2021, 10:59:53
Identifier
twxs.cmake
